x-pack/plugin/rollup/src/test/java/org/elasticsearch/xpack/rollup/v2/RollupActionSingleNodeTests.java

```diff
@@ -402,60 +502,173 @@ private RolloverResponse rollover(String dataStreamName) throws ExecutionExcepti
         return response;
     }
 
+    private Aggregations aggregate(final String index, AggregationBuilder aggregationBuilder) {
+        return client().prepareSearch(index).addAggregation(aggregationBuilder).get().getAggregations();
+    }
+
     @SuppressWarnings("unchecked")
-    private void assertRollupIndex(String sourceIndex, String rollupIndex, RollupActionConfig config) {
+    private void assertRollupIndex(String sourceIndex, String rollupIndex, RollupActionConfig config) throws IOException {
```

Comment:
This method is becoming too long. Is it worth breaking into separate steps such as `assertRollupSettings`, `assertRollupAggs`, `assertRollupMapping`. And then we can have `assertRollupIndex` call those methods